pkg/kubelet/cm/helpers.go

```diff
@@ -44,3 +51,33 @@ func hardEvictionReservation(thresholds []evictionapi.Threshold, capacity v1.Res
 	}
 	return ret
 }
+
+func buildContainerMapAndRunningSetFromRuntime(ctx context.Context, runtimeService internalapi.RuntimeService) (containermap.ContainerMap, sets.String) {
+	podSandboxMap := make(map[string]string)
+	podSandboxList, _ := runtimeService.ListPodSandbox(ctx, nil)
+	for _, p := range podSandboxList {
+		podSandboxMap[p.Id] = p.Metadata.Uid
+	}
+
+	runningSet := sets.NewString()
```

Comment:
According to the code-comments [1]:
```go
// Deprecated: use generic Set instead.
// new ways:
// s1 := Set[string]{}
// s2 := New[string]()
```

Would you mind switching to the generic Set that's not deprecated?

[1] https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go#L21

Reply:
Done, thanks

Reply:
The PR adding the generic set was introduced in 1.26. I'd like to have this PR potentially and easily backportable down to 1.25, because the regression we are fixing was caused by another fix backported to 1.25. For this reason I prefer to use the deprecated (but still valid) syntax. If we want a different PR specific for 1.25 and use the generics in 1.26 and onwards, I'm open to that as well.

Reply:
Yeah, makes total sense.
We can also defer this to a follow-up after everything is backported.

Thanks a lot!